_documentation/fresh-ubuntu-18.md

```diff
@@ -0,0 +1,239 @@
+---
+title: "Install Kimai on a fresh Ubuntu 18.04 LTS"
+description: "How to install Kimai 2 on a brand new Ubunutu 18.04 with database, webserver and SSL certificate"
+toc: true
+---
+
+This is a collection of snippets to help you with setting up a fresh Ubuntu 18.04 server for using with Kimai 2.
+It is neither a full fledged documentation, explaining each step, nor is it a bash tutorial. 
+
+Please see it as a personal snippet collection... in which I assume:
+- that you are familiar with the Linux bash and have at least basic knowledge of vim
+- that you use a single domain on this server, change the nginx configuration accordingly if you have multiple VirtualHosts
+- that you know how to protect your server (UFW, Fail2Ban ...) and can securely run it in the "wild"
+
+You must additionally:
+- replace `IP-of-myserver` with the server IP
+- replace the username `kevin` with your own
+- replace the domain `www.kimai.local` with your own
+
+## Accounts and SSH connection
+
+We start on our local machine, connect to the server and create our real user account:
+```bash
+ssh root@IP-of-myserver
+
+useradd -m -s /bin/bash kevin
+passwd kevin
+```
+
+Enable sudo access for this new user:
+```bash
+visudo /etc/sudoers.d/kevin
+```
+
+And paste this one line:
+```
+kevin   ALL=(ALL:ALL) ALL
+```
+
+Back to our local machine:
+```bash
+exit
+```
+
+Generate your SSH key and sent it to your server:
+```bash
+ssh-keygen -f ~/.ssh/kevin_myserver_rsa
+ssh-copy-id -i /Users/kevin/.ssh/kevin_myserver_rsa.pub kevin@IP-of-myserver
+```
+
+Then edit your local SSH config:
+```bash
+vim ~/.ssh/config
+```
+
+
+And paste this:
+```
+Host myserver
+    HostName IP-of-myserver
+    IdentityFile ~/.ssh/kevin_myserver_rsa
+    User kevin
+```
+
+And finally on to the server to start the software installation:
+
+```bash
+ssh myserver
+```
+
+### Secure your SSHD configuration
+
+Make sure your SSH server has at least some basic security settings in place:
+```bash
+sudo su
+vim /etc/ssh/sshd_config
+```
+
+Change those:
+```
+PermitRootLogin no
+PasswordAuthentication no
+```
+
+And restart the SSH Daemon:
+```bash
+/etc/init.d/ssh restart
+```
+
+## Install PHP, webserver and database
+
+Lets start with all required software:
+```bash
+apt-get update
```

Comment:
`apt-get` is the old form, nowadays simply `apt` is enough.

Reply:
Ha, didn't know that. Thanks for the info. Will try it next time

Reply:
You also get a nice progress bar with apt, so it's definitely better!